libvast/src/table_slice.cpp

```diff
@@ -740,24 +740,25 @@ filter(const table_slice& slice, expression expr, const ids& hints) {
   if (!hints.empty())
     selection &= hints;
   // Do no rows qualify?
-  auto selection_rank = rank(selection);
-  if (selection_rank == 0)
+  if (!any(selection))
```

Comment:
Both `rank` and `any` iterate over the bit ranges. If you switch back to `rank` here and at line 755 you can save the pass at line 759.

Reply:
I cannot, as `selection` may be updated by the call to `evaluate` in L753.

Doesn't `any` return as soon as it finds the first one bit, rather than iterating over the entire range?

Reply:
Ah, yes, of course it does.